full-service/src/db/wallet_db.rs

```diff
@@ -196,6 +197,35 @@ impl WalletDb {
         conn.batch_execute("PRAGMA foreign_keys = ON;")
             .expect("failed enabling foreign keys");
     }
+
+    pub fn run_proto_conversions_if_necessary(conn: &SqliteConnection) {
+        Self::run_assigned_subaddress_proto_conversions(conn);
+    }
+
+    fn run_assigned_subaddress_proto_conversions(conn: &SqliteConnection) {
+        let assigned_subaddresses = assigned_subaddresses::table
+            .load::<AssignedSubaddress>(conn)
+            .expect("failed querying for assigned subaddresses");
+
+        for assigned_subaddress in assigned_subaddresses {
+            let spend_public_key_bytes = match mc_util_serial::decode::<RistrettoPublic>(
+                &assigned_subaddress.spend_public_key,
+            ) {
+                Ok(spend_public_key) => spend_public_key.to_bytes().to_vec(),
+                Err(_) => continue,
```

Comment:
Well, there's no guarantee that these are protobuf or raw data bytes, and it's still important to check through all of them even if a subset are already converted. Why do you feel ```continue``` is dangerous?

Reply:
IIUC it would silently skip over records it cannot migrate, which would then get clobbered/lost. Would it make sense to log a warning or return an error?

Reply:
Yeah it's fine if it does that, IMO, it would skip over anything that is already converted, and there should _definitely_ not be anything that's corrupted